src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.c

```diff
@@ -61,25 +76,999 @@ static ColliderCylinderInit D_809D5C40 = {
+            }
+            Matrix_RotateZ(this->roll, MTXMODE_APPLY);
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx2->state.gfxCtx, "../z_en_clear_tag.c", 1104),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, gArwingShadowDL);
+        }
+    }
+
+    if (this->drawMode != CLEAR_TAG_DRAWMODE_MODEL) {
+        EnClearTag_DrawEffects(globalCtx);
+    }
+
+    CLOSE_DISPS(globalCtx2->state.gfxCtx, "../z_en_clear_tag.c", 1119);
+}
+
+/**
+ * Updates the Arwing effects.
+ * Performs effect physics.
+ * Moves and bounces debris effects.
+ * Fades most effects out of view. When effects are completely faded away they are removed.
+ */
+void EnClearTag_UpdateEffects(GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2;
+    s16 i;
+    f32 originalYPosition;
+    Vec3f sphereCenter;
+    EnClearTagEffect* effect;
+    s32 pad;
+
+    globalCtx2 = globalCtx;
+    effect = (EnClearTagEffect*)globalCtx2->specialEffects;
+
+    for (i = 0; i < CLEAR_TAG_EFFECT_MAX_COUNT; i++, effect++) {
+        if (effect->type != CLEAR_TAG_EFFECT_AVAILABLE) {
+            effect->random++;
+
+            // Perform effect physics.
+            effect->position.x += effect->velocity.x;
+            originalYPosition = effect->position.y;
+            effect->position.y += effect->velocity.y;
+            effect->position.z += effect->velocity.z;
+            effect->velocity.x += effect->acceleration.x;
+            effect->velocity.y += effect->acceleration.y;
+            effect->velocity.z += effect->acceleration.z;
+
+            if (effect->type == CLEAR_TAG_EFFECT_DEBRIS) {
+                // Clamp the velocity to -5.0
+                if (effect->velocity.y < -5.0f) {
+                    effect->velocity.y = -5.0f;
+                }
+
+                // While the effect is falling check if it has hit the ground.
+                if (effect->velocity.y < 0.0f) {
+                    sphereCenter = effect->position;
+                    sphereCenter.y += 5.0f;
+
+                    // Check if the debris has hit the ground.
+                    if (BgCheck_SphVsFirstPoly(&globalCtx2->colCtx, &sphereCenter, 11.0f)) {
+                        effect->position.y = originalYPosition;
+
+                        // Bounce the debris effect.
+                        if (effect->bounces <= 0) {
+                            effect->bounces++;
+                            effect->velocity.y *= -0.5f;
+                            effect->timer = ((s16)Rand_ZeroFloat(20)) + 25;
+                        } else {
+                            // The Debris effect is done bounding. Set it's velocity and acceleration to 0.
+                            effect->velocity.y = 0.0f;
+                            effect->acceleration.y = 0.0f;
+                            effect->velocity.z = 0.0f;
+                            effect->velocity.x = 0.0f;
```

Comment:
possible chained assigns, see if it can be done in `xyz` order.

Reply:
I had already tried this. Not OK.